### PR TITLE
Support match from id Set and fix some related issues

### PR DESCRIPTION
**Pull request by czpmango**
## What type of PR is this?
- [x] bug
- [x] feature
- [ ] enhancement

#### Description:
https://discuss.nebula-graph.com.cn/t/topic/12208

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [x] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [x] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

### CODE REVIEW

File: src/common/datatypes/Set.cpp [hunk 1]
```diff
@@ -11,6 +11,20 @@
 
 namespace nebula {
 
+Set Set::set_intersection(const Set& lhs, const Set& rhs) {
```
Comment (Aiee) [inline src/common/datatypes/Set.cpp:14]: Use camelcase naming.

Comment (czpmango) [reply src/common/datatypes/Set.cpp:14]: Consistent with std::set_intersection naming.

**Review by jievince — APPROVED** [review]

File: src/common/datatypes/Set.cpp (hunk 1 above)
Comment (Aiee) [reply src/common/datatypes/Set.cpp:14]: Let's keep the consistence in the project

**Review by yixinglu — APPROVED** [review]

**Comment by abby-cyber** [comment]
https://github.com/vesoft-inc/nebula-docs-cn/pull/2692

**Pull request merged**